File: invesdwin-context-parent/invesdwin-context-javascript/src/test/java/de/invesdwin/context/javascript/tests/callback/ParametersAndReturnsTestDecimal.js
print("getDecimal")
if(typeof getDecimal !== 'undefined')
	throw "getDecimal already defined!"
getDecimal = callback("getDecimal")
getDecimalType = typeof getDecimal
print(getDecimalType)
print(getDecimal)
if(getDecimalType !== 'number')
	throw "getDecimal not number!"
callback("setDecimal",getDecimal)

print("getDecimalVector")
if(typeof getDecimalVector !== 'undefined')
	throw "getDecimalVector already defined!"
getDecimalVector = callback("getDecimalVector")
getDecimalVectorType = typeof getDecimalVector[0]
print(getDecimalVectorType)
print(getDecimalVector)
if(getDecimalVectorType !== 'number')
	throw "getDecimalVector not number!"
callback("setDecimalVector",getDecimalVector)

print("getDecimalVectorAsList")
if(typeof getDecimalVectorAsList !== 'undefined')
	throw "getDecimalVectorAsList already defined!"
getDecimalVectorAsList = callback("getDecimalVectorAsList")
getDecimalVectorAsListType = typeof getDecimalVectorAsList[0]
print(getDecimalVectorAsListType)
print(getDecimalVectorAsList)
if(getDecimalVectorAsListType !== 'number')
	throw "getDecimalVectorAsList not number!"
callback("setDecimalVectorAsList",getDecimalVectorAsList)

print("getDecimalMatrix")
if(typeof getDecimalMatrix !== 'undefined')
	throw "getDecimalMatrix already defined!"
getDecimalMatrix = callback("getDecimalMatrix")
getDecimalMatrixType = typeof getDecimalMatrix[0][0]
print(getDecimalMatrixType)
print(getDecimalMatrix)
if(getDecimalMatrixType !== 'number')
	throw "getDecimalMatrix not number!"
callback("setDecimalMatrix",getDecimalMatrix)

print("getDecimalMatrixAsList")
if(typeof getDecimalMatrixAsList !== 'undefined')
	throw "getDecimalMatrixAsList already defined!"
getDecimalMatrixAsList = callback("getDecimalMatrixAsList")
getDecimalMatrixAsListType = typeof getDecimalMatrixAsList[0][0]
print(getDecimalMatrixAsListType)
print(getDecimalMatrixAsList)
if(getDecimalMatrixAsListType !== 'number')
	throw "getDecimalMatrixAsList not number!"
callback("setDecimalMatrixAsList",getDecimalMatrixAsList)

File: invesdwin-context-parent/invesdwin-context-jruby/src/test/java/de/invesdwin/context/jruby/tests/callback/ParametersAndReturnsTestDecimal.rb
puts("getDecimal")
unless (defined?(getDecimal)).nil? || getDecimal.nil?
	raise "getDecimal already defined!"
end
getDecimal = callback("getDecimal")
getDecimalType = getDecimal.class
puts(getDecimalType)
puts(getDecimal)
if(getDecimalType != Float)
	raise "getDecimal not Float!"
end
callback("setDecimal", getDecimal)

puts("getDecimalVector")
unless (defined?(getDecimalVector)).nil? || getDecimalVector.nil?
	raise "getDecimalVector already defined!"
end
getDecimalVector = callback("getDecimalVector")
getDecimalVectorType = getDecimalVector[0].class
puts(getDecimalVectorType)
puts(getDecimalVector)
if(getDecimalVectorType != Float)
	raise "getDecimalVector not Float!"
end
callback("setDecimalVector", getDecimalVector)

puts("getDecimalVectorAsList")
unless (defined?(getDecimalVectorAsList)).nil? || getDecimalVectorAsList.nil?
	raise "getDecimalVectorAsList already defined!"
end
getDecimalVectorAsList = callback("getDecimalVectorAsList")
getDecimalVectorAsListType = getDecimalVectorAsList[0].class
puts(getDecimalVectorAsListType)
puts(getDecimalVectorAsList)
if(getDecimalVectorAsListType != Float)
	raise "getDecimalVectorAsList not Float!"
end
callback("setDecimalVectorAsList", getDecimalVectorAsList)

puts("getDecimalMatrix")
unless (defined?(getDecimalMatrix)).nil? || getDecimalMatrix.nil?
	raise "getDecimalMatrix already defined!"
end
getDecimalMatrix = callback("getDecimalMatrix")
getDecimalMatrixType = getDecimalMatrix[0][0].class
puts(getDecimalMatrixType)
puts(getDecimalMatrix)
if(getDecimalMatrixType != Float)
	raise "getDecimalMatrix not Float!"
end
callback("setDecimalMatrix", getDecimalMatrix)

puts("getDecimalMatrixAsList")
unless (defined?(getDecimalMatrixAsList)).nil? || getDecimalMatrixAsList.nil?
	raise "getDecimalMatrixAsList already defined!"
end
getDecimalMatrixAsList = callback("getDecimalMatrixAsList")
getDecimalMatrixAsListType = getDecimalMatrixAsList[0][0].class
puts(getDecimalMatrixAsListType)
puts(getDecimalMatrixAsList)
if(getDecimalMatrixAsListType != Float)
	raise "getDecimalMatrixAsList not Float!"
end
callback("setDecimalMatrixAsList", getDecimalMatrixAsList)
